sympy/calculus/tests/test_util.py

```diff
@@ -31,3 +33,151 @@ def test_not_empty_in():
     assert not_empty_in(FiniteSet(1).intersect(Interval(3, 4)), x) == S.EmptySet
     assert not_empty_in(FiniteSet(x**2/(x + 2)).intersect(Interval(1, oo)), x) == \
         Union(Interval(-2, -1, True, False), Interval(2, oo))
+
+
+def test_Limits():
+    assert Limits(1, 2).delta == S(1)
+    assert Limits(1, 2).mid == S(3)/2
+
+    assert Limits(1, 1) == S(1)
+
+    assert Limits(1, 2) + 1 == Limits(2, 3)
+    assert 1 + Limits(1, 2) == Limits(2, 3)
+    assert Limits(1, 2) + Limits(2, 3) == Limits(3, 5)
+
+    assert -Limits(1, 2) == Limits(-2, -1)
+
+    assert Limits(1, 2) - 1 == Limits(0, 1)
+    assert 1 - Limits(1, 2) == Limits(-1, 0)
+    assert Limits(2, 3) - Limits(1, 2) == Limits(0, 2)
+
+    assert x + Limits(1, 2) == Add(Limits(1, 2), x)
+    assert a + Limits(1, 2) == Limits(1 + a, 2 + a)
+    assert Limits(1, 2) - x == Add(Limits(1, 2), -x)
+
+    assert Limits(-oo, 1) + oo == Limits(-oo, oo)
+    assert Limits(1, oo) + oo == oo
+    assert Limits(1, oo) - oo == Limits(-oo, oo)
+    assert (-oo - Limits(-1, oo)) == -oo
+    assert Limits(-oo, 1) - oo == -oo
+
+    assert Limits(1, oo) - oo == Limits(-oo, oo)
+    assert Limits(-oo, 1) - (-oo) == Limits(-oo, oo)
+    assert (oo - Limits(1, oo)) == Limits(-oo, oo)
+    assert (-oo - Limits(1, oo)) == -oo
+
+    assert Limits(1, 2)/2 == Limits(S(1)/2, 1)
+    assert 2/Limits(2, 3) == Limits(S(2)/3, 1)
+    assert 1/Limits(-1, 1) == Limits(-oo, oo)
+
+    assert abs(Limits(1, 2)) == Limits(1, 2)
+    assert abs(Limits(-2, -1)) == Limits(1, 2)
+    assert abs(Limits(-2, 1)) == Limits(0, 2)
+    assert abs(Limits(-1, 2)) == Limits(0, 2)
+
+
+def test_Limits_mul():
+    assert Limits(1, 2)*2 == Limits(2, 4)
+    assert 2*Limits(1, 2) == Limits(2, 4)
+    assert Limits(1, 2)*Limits(2, 3) == Limits(2, 6)
+
+    assert Limits(1, 2)*0 == 0
+    assert Limits(1, oo)*0 == Limits(0, oo)
+    assert Limits(-oo, 1)*0 == Limits(-oo, 0)
+    assert Limits(-oo, oo)*0 == Limits(-oo, oo)
+
+    assert Limits(1, 2)*x == Mul(Limits(1, 2), x, evaluate=False)
+
+    assert Limits(0, 2)*oo == Limits(0, oo)
+    assert Limits(-2, 0)*oo == Limits(-oo, 0)
+    assert Limits(0, 2)*(-oo) == Limits(-oo, 0)
+    assert Limits(-2, 0)*(-oo) == Limits(0, oo)
+    assert Limits(-1, 1)*oo == Limits(-oo, oo)
+    assert Limits(-1, 1)*(-oo) == Limits(-oo, oo)
+    assert Limits(-oo, oo) == Limits(-oo, oo)
+
+
+def test_Limits_div():
+    assert Limits(-1, 3)/Limits(3, 4) == Limits(-S(1)/3, 1)
+    assert Limits(-2, 4)/Limits(-3, 4) == Limits(-oo, oo)
+    assert Limits(-3, -2)/Limits(-4, 0) == Limits(S(1)/2, oo)
+#    assert Limits(-3, -2)/Limits(-2, 1) == Union(Limits(-oo, -2), Limits(1, oo))
```

Comment:
But a better answer is `Union(Limits(-oo, -2), Limits(1, oo))`. Here is proof of why  -1 ∉ `Limits(-3, -2)/Limits(-2, 1)`.
 Consider this `Limits(-3, -2)` is always `-ve`(or < 0). Hence for `Limits(-3, -2)/Limits(-2, 1)` to attain a value of `-1` the denominator must be `+ve`. So for `Limits(-2, 1)` we should consider only `Limits(0, 1)`(since `Limits(0, 1)` is +ve in this). Now only "minimum" `2` can satisfy `-2/2 = -1`. Hence -1 ∉ `Limits(-3, -2)/Limits(-2, 1)`.

On page 8, Theorem 7 of [this article](http://fab.cba.mit.edu/classes/S62.12/docs/Hickey_interval.pdf). I used that definition of `division` of `Limits`.